slave/src/main/java/org/fao/geonet/readonly/InterceptsStartup.java

```diff
@@ -0,0 +1,287 @@
+
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.domain.HarvesterSetting;
+import org.fao.geonet.domain.User;
+import org.fao.geonet.domain.User_;
+import org.fao.geonet.events.server.ServerStartup;
+import org.fao.geonet.kernel.harvest.HarvestManager;
+import org.fao.geonet.repository.HarvesterSettingRepository;
+import org.fao.geonet.repository.UserRepository;
+import org.jdom.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+/**
+ * 
+ * This instance is a slave, which means: No user login (no filter configured
+ * to authenticate) and no user creation {@link InterceptsUserAdd}
+ * {@link InterceptsUserUpdate}
+ * 
+ * On this class we will load a harvester defined by an url on a source file, if
+ * not already added, and remove all users except the admin user, just in case.
+ * 
+ * @author delawen
+ *
+ */
+@Component
+public class InterceptsStartup implements ApplicationListener<ServerStartup> {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	private @Value("${slave.removeHarvesters:true}") boolean removeHarvesters;
+	private @Value("${slave.removeUsers:true}") boolean removeUsers;
+	private @Value("${slave.masterURL}") String masterURL;
+	private @Value("${slave.masterNode}") String masterNode;
+	private @Value("${slave.frequency}") String frequency;
+
+	private final String key = "MASTER";
+	private final Logger log = LoggerFactory.getLogger(Geonet.GEONETWORK);
+
+	public void onApplicationEvent(ServerStartup arg0) {
+		removeHarvesters(arg0);
+		addHarvester(arg0);
+		removeUsers(arg0);
+	}
+
+	/**
+	 * Remove all harvesters that don't have the name "MASTER"
+	 * 
+	 * @param arg0
+	 */
+	@Transactional(value = TxType.REQUIRES_NEW)
+	private void removeHarvesters(ServerStartup arg0) {
+		if (removeHarvesters) {
+			HarvesterSettingRepository settingsRepo = arg0.getContext().getBean(HarvesterSettingRepository.class);
+
+			List<HarvesterSetting> harvesters = settingsRepo.findAllChildren(1);
+			for (HarvesterSetting h : harvesters) {
+				for (HarvesterSetting hs : settingsRepo.findAllChildren(h.getId())) {
+					if (hs.getName().equals("site")) {
+						for (HarvesterSetting s : settingsRepo.findAllChildren(hs.getId())) {
+							if (s.getName().equals("name") && !s.getValue().equals(key)) {
+								try {
+									HarvestManager hm = arg0.getContext().getBean(HarvestManager.class);
+									hm.remove(String.valueOf(h.getId()));
+								} catch (Exception e) {
+									log.error("Error removing old harvester in readonly instance: " + h, e);
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Remove all users with an ID above 1 (the default admin user we need to run
```

Comment:
It will fail if the default admin was dropped ? Should you consider to have a user id to preserve as config ?